ext/web/03_abort_signal.js

```diff
@@ -13,27 +14,115 @@ import {
 } from "ext:deno_web/02_event.js";
 const primordials = globalThis.__bootstrap.primordials;
 const {
+  ArrayPrototypeEvery,
+  ArrayPrototypePush,
   SafeArrayIterator,
   SafeSet,
   SafeSetIterator,
+  SafeWeakRef,
+  SafeWeakSet,
   SetPrototypeAdd,
   SetPrototypeDelete,
   Symbol,
   TypeError,
+  WeakRefPrototypeDeref,
+  WeakSetPrototypeAdd,
+  WeakSetPrototypeHas,
 } = primordials;
 import { refTimer, setTimeout, unrefTimer } from "ext:deno_web/02_timers.js";
 
+// Since WeakSet is not a iterable, WeakRefSet class is provided to store and
+// iterate objects.
+// To create an AsyncIterable using GeneratorFunction in the internal code,
+// there are many primordial considerations, so we simply implement the
+// toArray method.
+class WeakRefSet {
+  #weakSet = new SafeWeakSet();
+  #refs = [];
+
+  add(value) {
+    if (WeakSetPrototypeHas(this.#weakSet, value)) {
+      return;
+    }
+    WeakSetPrototypeAdd(this.#weakSet, value);
+    ArrayPrototypePush(this.#refs, new SafeWeakRef(value));
+  }
+
+  has(value) {
+    return WeakSetPrototypeHas(this.#weakSet, value);
+  }
+
+  toArray() {
+    const ret = [];
+    for (let i = 0; i < this.#refs.length; ++i) {
+      const value = WeakRefPrototypeDeref(this.#refs[i]);
+      if (value !== undefined) {
+        ArrayPrototypePush(ret, value);
+      }
+    }
+    return ret;
+  }
+}
+
 const add = Symbol("[[add]]");
 const signalAbort = Symbol("[[signalAbort]]");
 const remove = Symbol("[[remove]]");
 const abortReason = Symbol("[[abortReason]]");
 const abortAlgos = Symbol("[[abortAlgos]]");
+const dependent = Symbol("[[dependent]]");
+const sourceSignals = Symbol("[[sourceSignals]]");
+const dependentSignals = Symbol("[[dependentSignals]]");
 const signal = Symbol("[[signal]]");
 const timerId = Symbol("[[timerId]]");
 
 const illegalConstructorKey = Symbol("illegalConstructorKey");
 
 class AbortSignal extends EventTarget {
+  static any(signals) {
+    const prefix = "Failed to call 'AbortSignal.any'";
+    webidl.requiredArguments(arguments.length, 1, prefix);
```

Comment:
All the logic below this belongs to "create a dependent abort signal". It would be nice to have a link to https://dom.spec.whatwg.org/#create-a-dependent-abort-signal here.

Also, it would be good to split this logic to its own function, as it will be useful for other modules too. See for example step 30 of https://fetch.spec.whatwg.org/#dom-request.